custom_components/frank_energie/coordinator.py

```diff
@@ -94,3 +90,24 @@ async def _async_update_data(self) -> FrankEnergieData:
             DATA_MONTH_SUMMARY: data_month_summary,
             DATA_INVOICES: data_invoices,
         }
+
+    def __fetch_prices_with_fallback(self, start_date: date, end_date: date) -> MarketPrices:
+        if not self.api.is_authenticated:
+            return await self.api.prices(start_date, end_date)
+        else:
+            user_prices = await self.api.userPrices(start_date)
+
+            if len(user_prices.gas.all) > 0 and len(user_prices.electricity.all) > 0:
+                # If user_prices are available for both gas and electricity return them
+                return user_prices
+            else:
```

Comment:
Waarschijnlijk wel zo prettig om te loggen dat er gebruik wordt gemaakt van publieke prijsinformatie zodat dit voor de gebruiker zo transparant mogelijk is. (`LOGGER.info()`)